### PR TITLE
Application.properties per environment

### LINKED ISSUE #31: Define application.properties
It should exist one application.properties file per environment:
- development
- staging

Should contain necessary data to connect to postgres instance.

### PR DESCRIPTION
**Pull request by Ts1n**
- Created **application.properties** for each environment: local, staging and production
- Updated Dockerfile and buildDockerImage task to create the necessary environment variables such as:
    - `SPRING_PROFILES_ACTIVE` that will define the **application-{profile}.properties** to be used. Accepted values are **local**, **staging** and **production**.
    - Postgres credentials
    - I-On Core base url and token
- Updated workflows to inject values coming from Secrets
- Created `AppProperties` class to have typesafe acess to application properties

UPDATE:

- Docker container independent of environment
- `application.properties` expects environment variables to be provided by the running system

Closes #31

### CODE REVIEW

**Review by sscosta — APPROVED** [review]

**Review by sscosta — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -49,7 +50,13 @@ tasks.register<Exec>("buildDockerImage") {
     if (project.properties["onlyBuild"].toString().toBoolean()) {
         commandLine("docker", "build", ".")
     } else {
-        commandLine("docker", "build", ".", "--tag", "$tempDockerTag")
+        commandLine("docker", "build",
+            "--build-arg", "SPRING_PROFILE=${System.getenv()["SPRING_PROFILE"]}",
+            "--build-arg", "GOOGLE_CLOUD_SQL_POSTGRES_USER=${System.getenv()["GOOGLE_CLOUD_SQL_POSTGRES_USER"]}",
+            "--build-arg", "GOOGLE_CLOUD_SQL_POSTGRES_PASSWORD=${System.getenv()["GOOGLE_CLOUD_SQL_POSTGRES_PASSWORD"]}",
+            "--build-arg", "ION_CORE_BASE_URL=${System.getenv()["ION_CORE_BASE_URL"]}",
+            "--build-arg", "ION_CORE_TOKEN=${System.getenv()["ION_CORE_TOKEN"]}",
```
Comment (pmhsfelix) [inline build.gradle.kts]: This means we are creating docker images with secrets and environment specific information inside, right?
I don't like this approach. Would prefer images that are independent of the environment and don't have any secrets.

File: src/main/resources/application-local.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#spring
+spring.main.banner-mode=console
+
+#spring batch
+spring.datasource.driverClassName=org.postgresql.Driver
+spring.datasource.url=<POSTGRES-URL-GOES-HERE>
```
Comment (pmhsfelix) [inline src/main/resources/application-local.properties]: How is this replaced?

File: build.gradle.kts (hunk 1 above)
Comment (Ts1n) [reply build.gradle.kts]: How would we configure these properties that are specific to each environment (sta or prod)?

File: src/main/resources/application-local.properties (hunk 2 above)
Comment (Ts1n) [reply src/main/resources/application-local.properties]: This properties file is to be used locally and so the values are replaced by the user manually, according to how the environment was configured (datasource running locally or on cloud? I-on core should be running locally, but on what address/port?)

File: build.gradle.kts (hunk 1 above)
Comment (pmhsfelix) [reply build.gradle.kts]: *if possible* they should be provided by the environment and not by the application image, which _preferably_ should be environment independent.
If that is not possible due to our deployment model, I was wondering if we could instead create _app images_ (environment independent) and _environment images_ (on top of the _app images_), with the environment information

Comment (Ts1n) [reply build.gradle.kts]: The task `buildDockerImage` if called with `-PonlyBuild=true` just builds the docker image without any argument

**Review by pmhsfelix — APPROVED** [review]

**Pull request merged**